public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts

```diff
@@ -77,6 +79,8 @@ export default class Datasource extends DataSourceApi<AzureMonitorQuery, AzureDa
         continue;
       }
 
+      reportInteraction('grafana_query_sent', getAzureMonitorEvent(target));
```

Comment:
This is to show an alternative place where the logic could happen. But I don't think that's where it should be.

Pros:
* simplified: each plugin is responsible for their logging

Cons:
* logged each time the query is run (for dashboards set to auto-refresh every 10sec that's way too many!!!)
* we don't see queries that were set up but hidden in panel (although that's probably not a big deal)
* we can't use the event to answer proportion questions as we don't have access to the rest of the queries (e.g. number of dashboards that use azure Monitor as a datasource).

Reply:
> logged each time the query is run (for dashboards set to auto-refresh every 10sec that's way too many!!!)

Why is it too many? do we have any limitation?

But in any case, if you only want to report the event once per load, we could keep a local cache in the state that avoid reporting a query that has been already reported. But this may not be necessary if we can report it several times.

> we don't see queries that were set up but hidden in panel (although that's probably not a big deal)

I see that as benefit? we only count executed queries.

> we can't use the event to answer proportion questions as we don't have access to the rest of the queries (e.g. number of dashboards that use azure Monitor as a datasource).

I think it's important to define really well what you want to answer and do not try to foresee future questions.

Reply:
I also agree it's more scalable to not have each plugin doing the logging.